app/controllers/vm_common.rb

```diff
@@ -582,6 +582,16 @@ def snap_vm
         else
           render :action => "snap"
         end
+      elsif params[:description].blank? && @record.try(:snapshot_description_required?)
+        add_flash(_("Description is required"), :error)
+        @in_a_form = true
+        drop_breadcrumb(:description => _("Snapshot VM '%{description}'") % {:description => @record.description}, :url => "/vm_common/snap")
+        if session[:edit] && session[:edit][:explorer]
```

Comment:
Unfortunately this is not a hash its a Session object which does not support dig.